cmd/antrea-agent/options_linux.go

```diff
@@ -17,7 +17,234 @@
+	}
+
+	var ok bool
+	ok, o.trafficEncryptionMode = config.GetTrafficEncryptionModeFromStr(o.config.TrafficEncryptionMode)
+	if !ok {
+		return fmt.Errorf("TrafficEncryptionMode %s is unknown", o.config.TrafficEncryptionMode)
+	}
+	if o.trafficEncryptionMode == config.TrafficEncryptionModeNone && o.config.EnableIPSecTunnel {
+		klog.InfoS("enableIPSecTunnel is deprecated, use trafficEncryptionMode instead.")
+		o.trafficEncryptionMode = config.TrafficEncryptionModeIPSec
+	}
+
+	ok, o.trafficEncapMode = config.GetTrafficEncapModeFromStr(o.config.TrafficEncapMode)
+	if !ok {
+		return fmt.Errorf("TrafficEncapMode %s is unknown", o.config.TrafficEncapMode)
+	}
+
+	if o.trafficEncapMode.SupportsNoEncap() && o.trafficEncryptionMode != config.TrafficEncryptionModeNone {
+		return fmt.Errorf("TrafficEncryptionMode %s may only be enabled in %s mode", o.trafficEncryptionMode, config.TrafficEncapModeEncap)
+	}
+
+	if o.trafficEncryptionMode == config.TrafficEncryptionModeIPSec {
+		ok, o.ipsecAuthenticationMode = config.GetIPsecAuthenticationModeFromStr(o.config.IPsec.AuthenticationMode)
+		if !ok {
+			return fmt.Errorf("IPsec AuthenticationMode %s is unknown", o.config.IPsec.AuthenticationMode)
+		}
+
+		if o.ipsecAuthenticationMode == config.IPsecAuthenticationModeCert && !features.DefaultFeatureGate.Enabled(features.IPsecCertAuth) {
+			return fmt.Errorf("IPsec AuthenticationMode %s requires feature gate %s to be enabled", o.ipsecAuthenticationMode, features.IPsecCertAuth)
+		}
+	}
+
+	if o.config.NoSNAT && !(o.trafficEncapMode == config.TrafficEncapModeNoEncap || o.trafficEncapMode == config.TrafficEncapModeNetworkPolicyOnly) {
+		return fmt.Errorf("noSNAT is only applicable to the %s mode", config.TrafficEncapModeNoEncap)
+	}
+	if o.trafficEncapMode == config.TrafficEncapModeNetworkPolicyOnly {
+		// In the NetworkPolicyOnly mode, Antrea will not perform SNAT
+		// (but SNAT can be done by the primary CNI).
+		o.config.NoSNAT = true
+	}
+	return nil
+}
+
+func (o *Options) validateEgressConfig() error {
+	if !features.DefaultFeatureGate.Enabled(features.Egress) {
+		return nil
+	}
+	if o.trafficEncapMode != config.TrafficEncapModeEncap {
+		klog.InfoS("Egress requires 'encap' TrafficEncapMode, skipp running it")
+		return nil
+	}
+	if o.config.Egress.MaxEgressIPsPerNode > defaultMaxEgressIPsPerNode {
+		return fmt.Errorf("maxEgressIPsPerNode cannot be greater than %d", defaultMaxEgressIPsPerNode)
+	}
+
+	for _, cidr := range o.config.Egress.ExceptCIDRs {
+		_, ipNet, err := net.ParseCIDR(cidr)
+		if err != nil {
+			return fmt.Errorf("Egress Except CIDR %s is invalid", cidr)
+		}
+		o.exceptCIDRs = append(o.exceptCIDRs, ipNet)
+	}
+	o.enableEgress = true
+	return nil
+}
+
+func (o *Options) validateL7NetworkPolicyConfig() error {
+	if !features.DefaultFeatureGate.Enabled(features.L7NetworkPolicy) {
+		return nil
+	}
+	// L7NetworkPolicy depends on AntreaProxy.
+	if !o.enableAntreaPolicy {
```

Comment:
either the comment (AntreaProxy) is wrong or the condition is wrong (AntreaPolicy)